transport/http/outbound_test.go

```diff
@@ -453,3 +453,78 @@ func TestOutboundNoDeadline(t *testing.T) {
 	_, err := out.call(context.Background(), &transport.Request{})
 	assert.Equal(t, yarpcerrors.Newf(yarpcerrors.CodeInvalidArgument, "missing context deadline"), err)
 }
+
+func TestSvcMatchSuccess(t *testing.T) {
+	matchServer := httptest.NewServer(http.HandlerFunc(
+		func(w http.ResponseWriter, req *http.Request) {
+			defer req.Body.Close()
+			w.Header().Set(ServiceHeader, req.Header.Get(ServiceHeader))
+			_, err := w.Write([]byte("Service name header return"))
+			assert.NoError(t, err)
+		},
+	))
+	defer matchServer.Close()
+
+	httpTransport := NewTransport()
+	out := httpTransport.NewSingleOutbound(matchServer.URL)
+	require.NoError(t, out.Start(), "failed to start outbound")
+	defer out.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), testtime.Second)
+	defer cancel()
+	testSvcName := "Service"
+	_, err := out.Call(ctx, &transport.Request{
+		Service: testSvcName,
+	})
+	require.NoError(t, err)
+}
+
+func TestSvcMatchFailed(t *testing.T) {
+	dismatchServer := httptest.NewServer(http.HandlerFunc(
```

Comment:
spelling nit: replace all occurrences of `dismatch` with `mismatch`

Reply:
Resolved.